src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -21,6 +21,9 @@ public class CodeGenIL : IDisposable
         // True for internal methods (for example operators and unary operators)
         private IDictionary<string, IList> input;
         private IDictionary<string, IList> output;
+        /// <summary>
+        /// counter for local variables, should only be used directly during MethodLookup or GatherTypeInfo phase. It will incorrect during EmitIL phase. 
+        /// </summary>
```

Comment:
I think this can only be used in one phase. If it's used in `MethodLookup` phase, we'll be unable to use it in any subsequent phase, correct? Perhaps we should just mention that it's meant for use in `GatherTypeInfo` phase only.

Reply:
okay, yes, that is more clear.